.github/actions/publish_sns_topic/action.yaml

```diff
@@ -15,14 +15,9 @@ inputs:
 runs:
   using: composite
   steps:
-    - name: Mask topic ARN
-      run: echo "::add-mask::${{ inputs.sns_topic_arn }}"
-      shell: bash
-
     - name: Publish SNS topic
-      run: >
-        aws sns publish \
-          --topic-arn "${{ inputs.sns_topic_arn }}" \
-          --subject "${{ inputs.subject }}" \
-          --message "${{ inputs.body }}"
+      run: ./${{ github.action_path }}/publish.sh \
+          "${{ inputs.sns_topic_arn }}" \
+          "${{ inputs.subject }}" \
+          "${{ inputs.body }}"
```

Comment:
We want to reuse this SNS publishing logic when tests fail, but we can't dynamically spawn one or more actions from the context of a workflow (as far as I know), so I factored the logic out into a shared script instead.